compiler/rustc_interface/src/passes.rs

```diff
@@ -759,7 +759,9 @@ fn run_required_analyses(tcx: TyCtxt<'_>) {
     tcx.hir().par_body_owners(|def_id| {
         if tcx.is_coroutine(def_id.to_def_id()) {
             tcx.ensure().mir_coroutine_witnesses(def_id);
-            tcx.ensure().check_coroutine_obligations(def_id);
+            tcx.ensure().check_coroutine_obligations(
```

Comment:
Only thing I'm kinda unsatisfied with is calling `check_coroutine_obligations` once per coroutine rather than once per typeck root. Not sure if there's a better place to relocate this check, tho.